pkg/controllers/state/statenode.go

```diff
@@ -142,32 +154,37 @@ func (in *StateNode) Annotations() map[string]string {
 	// If the machine exists and the state node isn't initialized
 	// use the machine representation of the annotations
 	if in.Node == nil {
-		return in.Machine.Annotations
+		return in.NodeClaim.Annotations
 	}
-	if in.Machine == nil {
+	if in.NodeClaim == nil {
 		return in.Node.Annotations
 	}
 	// TODO @joinnis: The !in.Initialized() check can be removed when we can assume that all nodes have the v1alpha5.NodeRegisteredLabel on them
 	// We can assume that all nodes will have this label and no back-propagation will be required once we hit v1
 	if !in.Registered() && !in.Initialized() {
-		return in.Machine.Annotations
+		return in.NodeClaim.Annotations
 	}
 	return in.Node.Annotations
 }
 
+// GetLabels exists to conform to the GetLabels() function on the metav1.Object interface
```

Comment:
Where are we using this as a metav1.Object out of curiosity? Hard to find where, if you can just point me to it.

Reply:
It's called in [`OwnerKey`](https://github.com/jonathan-innis/karpenter-core/blob/provisioner-v1beta1-support/pkg/utils/nodeclaim/nodeclaim.go#L460) so that we can collapse getting the `OwnerKey` from both `runtime.Objects` (e.g. `Machines` and `NodeClaims`) and the `state.StateNode`